File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/CustModeTransaction.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Parameter for the CustMode/CalcTaxes and CustMode/CalcAdj APIs.
 */
public class CustModeTransaction {

    @SerializedName("CustBatchID")
    private Long custBatchID = null;

    @SerializedName("Transaction")
    private Transaction transaction = null;

    /**
     * Sets the customer batch identifier.
     * @param custBatchID
     * @return 
     */
    public CustModeTransaction custBatchID(Long custBatchID) {
        this.custBatchID = custBatchID;
        return this;
    }

    /**
     * Customer batch identifier.
     * @return custBatchID
     */
    public Long getCustBatchID() {
        return custBatchID;
    }

    /**
     * Sets the customer batch identifier.
     * @param custBatchID 
     */
    public void setCustBatchID(Long custBatchID) {
        this.custBatchID = custBatchID;
    }

    /**
     * Sets the transaction to include into the customer batch.
     * @param transaction
     * @return 
     */
    public CustModeTransaction transaction(Transaction transaction) {
        this.transaction = transaction;
        return this;
    }

    /**
     * Transaction to include into the customer batch.
     * @return transaction
     */
    public Transaction getTransaction() {
        return transaction;
    }

    /**
     * Sets the transaction to include into the customer batch.
     * @param transaction 
     */
    public void setTransaction(Transaction transaction) {
        this.transaction = transaction;
    }
}

File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/CustModeTransaction.js
/**
 * Parameter for the CustMode/CalcTaxes and CustMode/CalcAdj APIs.
 */

function CustModeTransaction() {

    this.CustBatchID = 0;
    this.Transaction = null;

    /**
     * Sets the customer batch identifier.
     * @param custBatchID Number
     * @return Reference to this object for method chaining. 
     */
    this.custBatchID = function (custBatchID) {
        this.CustBatchID = custBatchID;
        return this;
    }

    /**
     * Customer batch identifier.
     * @return Number
     */
    this.getCustBatchID = function () {
        return this.CustBatchID;
    }

    /* Reference to this object for method chaining.*
     * Sets the customer batch identifier.
     * @param custBatchID Number 
     */
    this.setCustBatchID = function (custBatchID) {
        this.CustBatchID = custBatchID;
    }

    /**
     * Sets the transaction to include into the customer batch.
     * @param transaction Transaction object.
     * @return Reference to this object for method chaining. 
     */
    this.transaction = function (transaction) {
        this.Transaction = transaction;
        return this;
    }

    /**
     * Transaction to include into the customer batch.
     * @return Transaction object.
     */
    this.getTransaction = function () {
        return this.Transaction;
    }

    /* Reference to this object for method chaining.*
     * Sets the transaction to include into the customer batch.
     * @param transaction Transaction object. 
     */
    this.setTransaction = function (transaction) {
        this.Transaction = transaction;
    }
}
